pkg/networking/cilium/reconciler/reconciler.go

```diff
@@ -148,6 +166,21 @@ func (r *Reconciler) upgrade(ctx context.Context, logger logr.Logger, client cli
 	return controller.Result{}, nil
 }
 
+func (r *Reconciler) upgradeConfig(ctx context.Context, logger logr.Logger, client client.Client, installation *cilium.Installation, spec *cluster.Spec) (controller.Result, error) {
```

Comment:
this doesn't need to return a result
It will also simplify the code in the main reconcile method